src/cloud/benchflow/sender/sender.go

```diff
@@ -153,51 +180,74 @@ func consumeFromTopic(t TransformerSetting) {
 						panic(err)
 						}
 					fmt.Println("Script "+s.Script+" processed")
-					launchAnalyserScript(msg.Trial_id, msg.Total_trials_num, t.Topic)
+					totalTrialsNum, _ := strconv.Atoi(msg.Total_trials_num)
+					launchAnalyserScripts(msg.Trial_id, totalTrialsNum, t.Topic)
 					}
+				consumer.CommitUpto(m)
 			}
-			waitGroup.Done()
+		consumer.Close()
+		waitGroup.Done()
 		}()
 }
 
-func launchAnalyserScript(trialID string, totalTrials int, req string) {
+func launchAnalyserScripts(trialID string, totalTrials int, req string) {
 	var scripts []AnalyserScript
 	for _, s := range c.AnalysersSettings {
```

Comment:
Why don't you build a map when you initialise the sender, so that then the access can be less complex?